homer/datadog_checks/homer/check.py

```diff
@@ -0,0 +1,112 @@
+        """
+
+        # validate instance configs
+        homer_service_checks = instance.get('homer_service_checks')
+        if homer_service_checks is None:
+            self.service_check(
+                "services_up", AgentCheck.WARNING, message='agent misconfiguration: homer_service_checks is not valid'
+            )
+            return None
+        homer_config_db = instance.get("homer_config_db")
+        if homer_config_db is None:
+            self.service_check(
+                "metrics_up", AgentCheck.WARNING, message='agent misconfiguration: homer_config_db is not valid'
+            )
+            return None
+        homer_data_db = instance.get("homer_data_db")
+        if homer_data_db is None:
+            self.service_check(
+                "metrics_up", AgentCheck.WARNING, message='agent misconfiguration: homer_data_db is not valid'
+            )
+            return None
+
+        # check if homer processes are alive
+        down_services = []
+        for service in homer_service_checks:
+            if not HomerCheck.isProcAlive(service):
+                down_services.append(service)
+        if len(down_services) != 0:
+            self.service_check(
+                "services_up", AgentCheck.CRITICAL, message=f'found homer services that are down: {down_services}'
+            )
+            return None
+
+        self.service_check("services_up", AgentCheck.OK, message='homer services are all up')
+
+        # gather and send metrics
+        conn1 = ConnNoop()
+        conn2 = ConnNoop()
+        try:
+            conn1 = psycopg2.connect(
+                user=homer_config_db['user'],
+                password=homer_config_db['pass'],
+                host=homer_config_db['host'],
+                port=homer_config_db['port'],
+                dbname=homer_config_db['name'],
+            )
+            conn2 = psycopg2.connect(
+                user=homer_data_db['user'],
+                password=homer_data_db['pass'],
+                host=homer_data_db['host'],
+                port=homer_data_db['port'],
+                dbname=homer_data_db['name'],
+            )
+            with conn1.cursor() as cur1, conn2.cursor() as cur2:
+                cur1.execute("SELECT hepid, profile FROM mapping_schema WHERE hep_alias='SIP';")
+                for row in cur1:
+                    cur2.execute(
+                        "SELECT SUM(n_live_tup) FROM pg_stat_user_tables WHERE relname LIKE CONCAT('hep_proto', '_', %s, '_', %s, '_', TO_CHAR(CURRENT_DATE, 'YYYYMMDD'), '_%%');",  # noqa: E501
+                        (row[0], row[1]),
+                    )
+                    stat = cur2.fetchone()[0]
+                    self.gauge(f'txcount.{row[1]}', float(stat))
+        except Exception as ex:
+            self.service_check("metrics_up", AgentCheck.WARNING, message=f'error retrieving metrics: {str(ex)}')
+            return None
+        finally:
+            conn1.close()
+            conn2.close()
```

Comment:
Alternatively, you could remove `ConnNoop()` and close `conn1` and `conn2` if they are not None